Foundation/NSProcessInfo.swift

```diff
@@ -102,7 +102,18 @@ public class NSProcessInfo : NSObject {
     
     public var operatingSystemVersion: NSOperatingSystemVersion {
         get {
-            NSUnimplemented()
+            let productVersionKey = unsafeBitCast(_kCFSystemVersionProductVersionKey, UnsafePointer<Void>.self)
+            let productVersion = unsafeBitCast(CFDictionaryGetValue(_CFCopySystemVersionDictionary(), productVersionKey), NSString!.self)
+            if productVersion == nil {
+                fatalError("Could not read operating system version.")
+            }
+            let versionComponents = productVersion._swiftObject.characters.split(".").flatMap(String.init).flatMap({ Int($0) })
+            guard versionComponents.count >= 3 else {
+                fatalError("Invalid operating system version")
```

Comment:
Here's what Darwin does in this case, and I think we should use it instead of a fatalError in either the null result case for Linux or count <3:

majorVersion = result or -1
minorVersion = result or 0
patchVersion = result or 0